app/controllers/flows_controller.rb

```diff
@@ -111,20 +111,17 @@ def self.for(type, controller)
           controller_list: Navigation::DiyNavigation::FLOW,
           form: nil
         )
-      when :state_file_az
-        FlowParams.new(
-          controller: controller,
-          reference_object: controller.current_intake&.is_a?(StateFileAzIntake) ? controller.current_intake : nil,
-          controller_list: Navigation::StateFileAzQuestionNavigation::FLOW,
-          form: SampleStateFileIntakeGenerator.new('az').form
-        )
-      when :state_file_ny
-        FlowParams.new(
-          controller: controller,
-          reference_object: controller.current_intake&.is_a?(StateFileNyIntake) ? controller.current_intake : nil,
-          controller_list: Navigation::StateFileNyQuestionNavigation::FLOW,
-          form: SampleStateFileIntakeGenerator.new('ny').form
-        )
+      else
+        if type.to_s.starts_with?('state_file_')
+          state_code = type.to_s.split('_').last
+          intake_class = StateFile::StateInformationService.intake_class_from_state_code(state_code)
+          FlowParams.new(
+            controller: controller,
+            reference_object: controller.current_intake&.is_a?(intake_class) ? controller.current_intake : nil,
+            controller_list: StateFile::StateInformationService.navigation_from_state_code(state_code)::FLOW,
```

Comment:
Since the only difference between each state is literally the state code, I ask that we not to store this in the `StateInformationService`, and instead derive it on the spot. I don't think it's a good idea to add too much to the service if it can otherwise be derived.

```suggestion
            controller_list: "Navigation::StateFile#{state_code.capitalize}QuestionNavigation".constantize::FLOW,
```

I also like the added benefit for enforcing a naming convention for `StateFileXxQuestionNavigation` classes.

Reply:
fine by me, and does seem good to enforce a naming convention

Reply:
per the slack thread, i forgot we had talked about constantize earlier. is there a way to make sure we don’t end up in a situation where injected parameters are a concern? if state code is often coming in via the url that seems like enough reason to just avoid doing it but i def don't feel like an expert here

Reply:
I added a static method that checks that the code is in the whitelist. Using this everywhere is preferable to peppering the code with constantize methods